ably-ios/ARTAuth.m

```diff
@@ -172,51 +190,77 @@ - (instancetype)initWithRest:(ARTRest *)rest options:(ARTAuthOptions *)options {
         _tokenCbs = nil;
         _tokenRequest = nil;
 
-        if (nil != options.keyValue && nil == options.clientId) {
-            _authMethod = ARTAuthMethodBasic;
+        
+        
+        if (nil != options.keyValue /*&& nil == options.clientId*/) {
```

Comment:
Why is `nil == options.clientId` commented out.  If a `clientId` is specified, then token authentication must be used. See https://github.com/ably/ably-java/blob/master/src/io/ably/rest/Auth.java#L588-L598